The job **`pr-check-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/24203485840/job/70653154868) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/70653154868) [(plain)](https://github.com/rust-lang/rust/commit/bc1203793624360b305783308d0711c40ea5e264/checks/70653154868/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] addr2line test:false 1.177
[RUSTC-TIMING] object test:false 11.113
warning: dropping unsupported crate type `dylib` for target `wasm32-wasip1`

error[E0425]: cannot find value `_SC_NPROCESSORS_ONLN` in crate `libc`
   --> library/std/src/sys/thread/unix.rs:183:48
    |
183 |             match unsafe { libc::sysconf(libc::_SC_NPROCESSORS_ONLN) } {
    |                                                ^^^^^^^^^^^^^^^^^^^^ not found in `libc`

For more information about this error, try `rustc --explain E0425`.
[RUSTC-TIMING] std test:false 4.278
warning: `std` (lib) generated 1 warning
```

</details>

--- a/library/std/src/sys/thread/unix.rs
+++ b/library/std/src/sys/thread/unix.rs
@@ -155,6 +155,7 @@ pub fn available_parallelism() -> io::Result<NonZero<usize>> {
             target_os = "aix",
             target_vendor = "apple",
             target_os = "cygwin",
+            target_os = "wasi",
         ) => {
             #[allow(unused_assignments)]
             #[allow(unused_mut)]

--- a/library/std/tests/thread.rs
+++ b/library/std/tests/thread.rs
@@ -85,7 +85,6 @@ fn thread_local_hygeiene() {
         target_env = "sgx",
         target_os = "solid_asp3",
         target_os = "teeos",
-        target_os = "wasi"
     ),
     should_panic
 )]

